mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/proto/UpgradeClusterContainersRequest.java

```diff
@@ -36,4 +37,9 @@ public class UpgradeClusterContainersRequest {
     int optionalBatchMaxSize;
 
     boolean forceUpgradeOnSameImage;
+
+    boolean enableSkuSpecUpgrade;
+
+    @Nullable
+    MantisResourceClusterSpec resourceClusterSpec;
```

Comment:
nit: why do we need this here? Do we not have a separate endpoint to update the cluster spec?

Reply:
I was thinking along the lines of:
```
resourceCluster.sku(SKU).clusterSpec(NEW_SPEC).image(IMAGE).reconcile()
```

Reply:
this is served as translation between actors to temporal actually as Temporal workflow. It's probably cleaner to have another intermediate contract class instead.